src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -26,14 +25,37 @@ object Lorre extends App with TezosErrors with LazyLogging {
   //how long to wait for graceful shutdown of system components
   private[this] val shutdownWait = 10.seconds
 
-  private val network =
-    if (args.length > 0) args(0)
-    else {
-      Console.err.println("""
-      | No tezos network was provided to connect to
-      | Please provide a valid network as an argument to the command line""".stripMargin)
-      sys.exit(1)
+  sealed class Depth()
+  case class Everything() extends Depth
+  case class Newest() extends Depth
+  case class Custom(depth: Int) extends Depth
+
+  case class Config(private val d: Int = -1, networks: Seq[String] = Seq()) {
+    def depth = {
+      d match {
```

Comment:
What about using a string to allow also
- `depth=new`
- `depth=all`
- `depth=42`
?